### ISSUE TITLE
Error trying to load auth webview activity with application context

### ISSUE THREAD

**Issue by elkhoudiry** [auth0/Auth0.Android#521]
Basically this is my setup, i'm using `ViewModel` which has a `login` function that calls `WebAuthProvider`
also using a dagger hilt provider for providing app level dependencies  for `Auth0`, `AuthenticationAPIClient`, etc

Everything is working alright and clean , i can call `login()` function with no paramaters and dagger hilt injects the dependencies for me, but passing context to login `WebAuthProvider` is a challenge
When i'm leaving the context to dagger hilt so it can inject an application context i get this error 

```
Calling startActivity() from outside of an Activity  context requires the  
FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
```

https://github.com/auth0/Auth0.Android/blob/9bbee00e09933c7c9b741ffe0fdb3a29385cce2a/auth0/src/main/java/com/auth0/android/provider/AuthenticationActivity.kt#L101

can this flag `FLAG_ACTIVITY_NEW_TASK ` be added to auth0 sdk ?

**Comment by lbalmaceda** [comment]
This has already been reported here: https://github.com/auth0/Auth0.Android/issues/479

The `WebAuthProvider` takes an `Activity` context rather than an `Application` context. But changing the signature today poses a breaking change. If you want to use the class in that use case, you will have to pass an activity.

**Comment by elkhoudiry** [comment]
thank you for your response, is it possible to be allowed to pass custom defined `intent` or something to overcome this application context problem ?

**Comment by lbalmaceda** [comment]
@elkhoudiry If your intention is to pass `FLAG_ACTIVITY_NEW_TASK`, that has proven to be problematic for the web auth scenario; as the custom tabs would be opened in a separate task defeating the purpose of their improved UX. See my comment in the issue linked in my previous comment.

Moreover, we are considering for future implementation of the WebAuthProvider to receive a `ComponentActivity` as context instead and use the `Activity Results API` to register that for an internal result. That change wouldn't be possible if we open the API to receive a custom intent.

**Comment by thsaraiva** [comment]
Hey @elkhoudiry , @lbalmaceda , I've reported this "not found" behaviour here:

https://community.auth0.com/t/google-sign-in-integration-in-android-app-simply-stopped-working/70975

but found a solution in stack overflow this morning:

https://stackoverflow.com/questions/2665796/android-handle-oauth-callback-using-intent-filter

TL;DR: change the scheme from `https://` to a custom scheme `your_company_name://`.

It would be interesting to understand exactly what is happening and why it used to work with `https` and it doesn't anymore.
